### PR TITLE
In Merge validate nkey and subsequent saveIfNewer error on invalid jwt

### PR DESCRIPTION
**Pull request by matthiashanel**
Signed-off-by: Matthias Hanel <mh@synadia.com>

 - [ ] Link to issue, e.g. `Resolves #NNN`
 - [ ] Documentation added (if applicable)
 - [ ] Tests added
 - [ ] Branch rebased on top of current main (`git pull --rebase origin main`)
 - [ ] Changes squashed to a single commit (described [here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
 - [ ] Build is green in Travis CI
 - [ ] You have certified that the contribution is your original work and that you license the work to the project under the [Apache 2 license](https://github.com/nats-io/nats-server/blob/main/LICENSE)

Resolves #

### Changes proposed in this pull request:

 -
 -
 -

/cc @nats-io/core

### CODE REVIEW

**Review by philpennock — COMMENTED** [review]
Is dirstore not compatible with non-JWT users?  Can't people use a cluster of servers configured with static users, and JS?  I don't think this is a safe change.

**Comment by matthiashanel** [comment]
> Can't people use a cluster of servers configured with static users, and JS? I don't think this is a safe change.

If they use a cluster of servers with static users and JS they won't be using this. 
this is exclusively used as backing storage by nats account resolver.
This was even done before JS went clustreed.

**Review by kozlovic — APPROVED** [review]
LGTM

**Comment by philpennock** [comment]
Oh sorry, dirstore.go vs filestore.go ... my head went in completely the wrong direction.

File: server/dirstore.go [hunk 1]
```diff
@@ -321,6 +323,9 @@ func (store *DirJWTStore) Merge(pack string) error {
 			return fmt.Errorf("line in package didn't contain 2 entries: %q", line)
 		}
 		pubKey := split[0]
+		if !nkeys.IsValidPublicAccountKey(pubKey) {
+			return fmt.Errorf("key to merge is not a valid public account key")
```
Comment (philpennock) [inline server/dirstore.go:327]: without a %param this should probably be `errors.New()` instead, but %q auto-quoting of the key might be worthwhile.

**Review by philpennock — APPROVED** [review]
Double-check with Alberto please: there are I think some assumptions about propagation of the JWTs across JWT version boundaries, and this change seems to change the semantics of how a future v3 JWT could be introduced, with every nats-server in a cluster having to be recompiled with a jwt library which accepts the format, before the JWTs can be propagated.

**Pull request merged**